js/renovation/ui/scroll_view.tsx

```diff
@@ -4,34 +4,171 @@ import {
   JSXComponent,
   Slot,
   OneWay,
+  Method,
+  Ref,
 } from 'devextreme-generator/component_declaration/common';
+import { isNumeric } from '../../core/utils/type';
+import { Widget } from './common/widget';
+import config from '../../core/config';
 
-export const viewFunction = (viewModel: ScrollView) => (
-  <div className={viewModel.cssClasses}>
+const DIRECTION_VERTICAL = 'vertical';
+const DIRECTION_HORIZONTAL = 'horizontal';
+const DIRECTION_BOTH = 'both';
+const SCROLLABLE_CONTENT_CLASS = 'dx-scrollable-content';
+
+export interface Location {
+  top: number;
+  left: number;
+}
+
+export const ensureLocation = (location: number | Location): Location => {
+  if (isNumeric(location)) {
+    return {
+      left: location,
+      top: location,
+    } as Location;
+  }
+  return { ...{ top: 0, left: 0 }, ...(location as Location) } as Location;
+};
+
+export const getRelativeLocation = (element: HTMLElement): Location => {
+  const result = { top: 0, left: 0 } as Location;
+  let targetElement = element;
+  while (!targetElement.matches(`.${SCROLLABLE_CONTENT_CLASS}`)) {
+    result.top += targetElement.offsetTop;
+    result.left += targetElement.offsetLeft;
+    targetElement = targetElement.offsetParent as HTMLElement;
+  }
+  return result;
+};
+
+export const viewFunction = ({
+  cssClasses, contentRef, containerRef,
+  props: {
+    disabled, height, width, rtlEnabled, children,
+  },
+}: ScrollView): JSX.Element => (
+  <Widget
```

Comment:
Maybe you missed restAttributes spreading

Reply:
I am fix it